core/Symbols.cc

```diff
@@ -1516,6 +1517,27 @@ TypePtr Symbol::sealedSubclassesToUnion(const GlobalState &gs) const {
     return result;
 }
 
+bool Symbol::hasSingleSealedSubclass(const GlobalState &gs) const {
+    ENFORCE(this->isClassOrModuleSealed(), "Class is not marked sealed: {}", ref(gs).show(gs));
+
+    auto sealedSubclasses = this->lookupSingletonClass(gs).data(gs)->findMember(gs, core::Names::sealedSubclasses());
+
+    // When the sealed type is a class, it must also be abstract for there to be a single subclass.
+    if (this->isClassOrModuleClass() && !this->isClassOrModuleAbstract()) {
+        return false;
+    }
```

Comment:
I think you actually want "either `module` or `abstract!` `class`".

https://github.com/Morriar/sorbet/blob/a76aae203375ff538a9d51e4fa03be587b1316b2/core/types/types.cc#L195-L199

^ Here is the condition we use elsewhere.

Reply:
(You might want to add a test for this.)

Reply:
Ah I see, i misread the condition. I think that the condition in this PR and the condition in the linked snippet are the same, but De Morgan'd